### PR TITLE
Fix donbot description

### PR DESCRIPTION
**Pull request by HoangGiang93**

### CODE REVIEW

File: iai_boxy_base_2/robots/base.xacro [hunk 1]
```diff
@@ -0,0 +1,149 @@
+<?xml version="1.0"?>
```
Comment (ichumuh) [inline iai_boxy_base_2/robots/base.xacro:1]: You renamed base.URDF to base.xacro? Why did you do that and why is the new file shorter? Other robot urdfs are using this one as well, so the risk of breaking the other urdfs has to be worth the gain from changing it.

File: iai_boxy_base_2/robots/base.URDF.xacro [hunk 2]
```diff
@@ -1 +0,0 @@
-base.URDF
```
Comment (ichumuh) [inline iai_boxy_base_2/robots/base.URDF.xacro:1]: Why did you delete this file?

File: iai_donbot_description/launch/display.launch [hunk 3]
```diff
@@ -1,22 +1,10 @@
...
-    pkg="rviz"
-    type="rviz"
-    args="-d $(find iai_boxy_base_2)/urdf.rviz" /-->
-</launch>
+	<arg name="model" />
+	<arg name="gui" default="True" />
+	<include file="$(find iai_donbot_description)/launch/upload.launch" />
+	<node pkg="tf2_ros" type="static_transform_publisher" name="fake_localization" output="screen" args="0 0 0 0 0 0 map odom" />
+	<param name="use_gui" value="$(arg gui)" />
+	<node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" />
```
Comment (ichumuh) [inline iai_donbot_description/launch/display.launch:7]: Why are you using joint_state_publisher_gui, instead of joint_state_publisher?

File: iai_donbot_description/robots/iai_donbot.xacro [hunk 4]
```diff
@@ -1,279 +1,239 @@
 <?xml version="1.0"?>
```
Comment (ichumuh) [inline iai_donbot_description/robots/iai_donbot.xacro:1]: There are a lot of changes here. Can you please describe whats going on?

File: iai_donbot_description/robots/iai_donbot_spawn.xacro [hunk 5]
```diff
@@ -5,7 +5,7 @@
```
Comment (ichumuh) [inline iai_donbot_description/robots/iai_donbot_spawn.xacro:5]: I would suggest calling this file `iai_donbot.urdf.xacro`.

**Review by ichumuh — CHANGES_REQUESTED** [review]

File: iai_boxy_base_2/robots/base.URDF.xacro (hunk 2 above)
Comment (HoangGiang93) [reply iai_boxy_base_2/robots/base.URDF.xacro:1]: Because every files are xacro, I want to make a base file that conclude option (with or without odom). So it must be a xacro

File: iai_boxy_base_2/robots/base.xacro (hunk 1 above)
Comment (HoangGiang93) [reply iai_boxy_base_2/robots/base.xacro:1]: It is shorter because I used an auto-format, the results are still the same. The different is now we have an option is to use or not to use odom.
So instead of writing  
<xacro:base/>
we write
<xacro:base odom="True"/>

File: iai_donbot_description/launch/display.launch (hunk 3 above)
Comment (HoangGiang93) [reply iai_donbot_description/launch/display.launch:7]: joint_state_publisher is deprecated, it shows up warning that I dont like

File: iai_donbot_description/robots/iai_donbot_spawn.xacro (hunk 5 above)
Comment (HoangGiang93) [reply iai_donbot_description/robots/iai_donbot_spawn.xacro:5]: I saw that kind of naming sometimes but still don't get why. Can you explain it to me?

File: iai_donbot_description/robots/iai_donbot.xacro (hunk 4 above)
Comment (HoangGiang93) [reply iai_donbot_description/robots/iai_donbot.xacro:1]: sorry because of my auto-format the tab-space is increase from 2 to 4, it looks like I changed everything but it wasn't. I can change it to 2 if necessary. The code is the same, I just did the following things:
1) Remove gazebo related nodes
2) Make it a macro, so we can call it as <xacro:iai_donbot odom="True"/>
I still keep the old comments, I think we will need to remove them if possible

File: iai_donbot_description/robots/iai_donbot_spawn.xacro (hunk 5 above)
Comment (HoangGiang93) [reply iai_donbot_description/robots/iai_donbot_spawn.xacro:5]: I changed it

File: iai_donbot_description/robots/iai_donbot.xacro (hunk 4 above)
Comment (ichumuh) [reply iai_donbot_description/robots/iai_donbot.xacro]: 4 would be fine, but if I count correctly, its even 8 at some places. I'd prefer if you'd change it back to 2. Same for iai boxy base 2.

File: iai_boxy_base_2/robots/base.URDF.xacro (hunk 2 above)
Comment (ichumuh) [reply iai_boxy_base_2/robots/base.URDF.xacro:1]: Have you made sure that this file is not used anywhere in this repo?

File: iai_donbot_description/robots/iai_donbot_spawn.xacro (hunk 5 above)
Comment (ichumuh) [reply iai_donbot_description/robots/iai_donbot_spawn.xacro:5]: Its to indicate that if you run `xacro iai_donbot.urdf.xacro`, the result will be a proper urdf.

File: iai_donbot_description/robots/iai_donbot.urdf.xacro [hunk 6]
```diff
@@ -107,7 +107,13 @@
...
 			<origin xyz="0 0 0" rpy="0 0 ${pi}" />
 		</joint>
-		<link name="camera_link" />
+		<link name="camera_link">
+			<visual>
+				<geometry>
+					<cylinder radius="0.001" length="0.005"/>
+				</geometry>
+			</visual>
+		</link>
```
Comment (ichumuh) [inline iai_donbot_description/robots/iai_donbot.urdf.xacro:116]: Why have you added a small cylinder to the camera link?

File: iai_donbot_description/robots/iai_donbot_spawn.xacro (hunk 5 above)
Comment (ichumuh) [reply iai_donbot_description/robots/iai_donbot_spawn.xacro:5]: I meant that iai_donbot_spawn.xacro gets renamed to iai_donbot.urdf.xacro. and iai_donbot.xacro keeps its name.

File: iai_donbot_description/robots/iai_donbot_with_refills_finger.xacro [hunk 7]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
```
Comment (ichumuh) [inline iai_donbot_description/robots/iai_donbot_with_refills_finger.xacro:1]: This file should probably be called `iai_donbot_with_refills_finger.urdf.xacro` as well.

**Review by ichumuh — CHANGES_REQUESTED** [review]

File: iai_boxy_base_2/robots/base.URDF.xacro (hunk 2 above)
Comment (HoangGiang93) [reply iai_boxy_base_2/robots/base.URDF.xacro:1]: I just checked it, this file is not used anywhere else

File: iai_donbot_description/robots/iai_donbot.xacro (hunk 4 above)
Comment (HoangGiang93) [reply iai_donbot_description/robots/iai_donbot.xacro:1]: I changed it back to 2

File: iai_donbot_description/robots/iai_donbot_spawn.xacro (hunk 5 above)
Comment (HoangGiang93) [reply iai_donbot_description/robots/iai_donbot_spawn.xacro]: ok I see

File: iai_donbot_description/robots/iai_donbot.urdf.xacro (hunk 6 above)
Comment (HoangGiang93) [reply iai_donbot_description/robots/iai_donbot.urdf.xacro]: It's for unreal to recognize this link, link without visual or collision will be ignored. I need this link for camera frame

File: iai_donbot_description/robots/iai_donbot_with_refills_finger.xacro (hunk 7 above)
Comment (HoangGiang93) [reply iai_donbot_description/robots/iai_donbot_with_refills_finger.xacro]: ok I changed it

**Comment by ichumuh** [comment]
tested on real donbot

**Comment by gaya-** [comment]
Does this change boxy's urdf?

**Comment by ichumuh** [comment]
no. apperently boxy uses iai_boxy_base instead of iai_boxy_base2, so the only overlap is the gripper, which didn't get changed. I've also checked the boxy simulator and it still works.

**Comment by gaya-** [comment]
@ichumuh thanks for double checking. @artnie fyi, you might want to avoid pulling iai_robots on boxy in any case.
